docs/src/main/sphinx/security/https.rst

```diff
@@ -0,0 +1,352 @@
+The PEM format is preferred, and is easier to use.
+
+Make sure you obtain a certificate that is certified by a recognized
+certification authority (CA), or that you submit a self-generated certificate
+to a CA for validation.
+
+.. _inspect-pems:
+
+Inspect PEM certificates
+""""""""""""""""""""""""
+
+When you receive your PEM format certificate, inspect it and verify that it
+shows the correct information for your Trino server. First, use the ``cat``
+command to view this plain text file. It should contain sections for a PRIVATE
+KEY and at least one CERTIFICATE:
+
+.. code-block:: text
+
+  -----BEGIN PRIVATE KEY-----
+  MIIEowIBAAKCAQEAwJL8CLeDFAHhZe3QOOF1vWt4Vuk9vyO38Y1y9SgBfB02b2jW
+  ....
+  -----END PRIVATE KEY-----
+  -----BEGIN CERTIFICATE-----
+  MIIDujCCAqICAQEwDQYJKoZIhvcNAQEFBQAwgaIxCzAJBgNVBAYTAlVTMRYwFAYD
+  ....
+  -----END CERTIFICATE-----
+  -----BEGIN CERTIFICATE-----
+  MIIDwjCCAqoCCQCxyqwZ9GK50jANBgkqhkiG9w0BAQsFADCBojELMAkGA1UEBhMC
+  ....
+  -----END CERTIFICATE-----
+
+If your PEM or key file reports ``BEGIN ENCRYPTED PRIVATE KEY``, you must use a
+password when invoking that key.
+
+If you received separate PEM files, one for the server's key and another for its
+certificate, concatenate the files into one, in order from key to cert.
+
+.. code-block:: text
+
+  cat key.pem cert.pem > server.pem
+
+.. _validate-pems:
+
+Validate key
+""""""""""""
+
+This page presumes your system uses OpenSSL 1.1 or later.
+
+Test the key's validity with the following command:
+
+.. code-block:: text
+
+  openssl rsa -in server.pem -check -noout
+
+Look for the following confirmation message:
+
+.. code-block:: text
+
+  RSA key ok
+
+Validate certificate
+""""""""""""""""""""
+
+For the cert file, analyze it with a different ``openssl`` command:
+
+.. code-block:: text
+
+  openssl x509 -in server.pem -text -noout
+
+If your cert was generated with a password, ``openssl`` prompts for it.
+
+In the output of the ``openssl`` command, look for the following
+characteristics:
+
+* Modern browsers now enforce 398 days as the maximum validity period for a
+  cert. Look for ``Not Before`` and ``Not After`` dates in the ``Validity``
+  section of the output, and make sure the time span does not exceed 398
+  days.
+* If you received an x509 version 3 certificate, it may have a **Subject
+  Alternative Name** field. If present, make sure this shows the DNS name of
+  your server, such as ``DNS:trino.example.com``
+* The legacy common name (CN) field is ignored by modern browsers, but is a good
+  visual aid to distinguish certs. Example: ``CN=trino.example.com``
+
+Invalid certs
+"""""""""""""
+
+If your cert does not pass validation, or does not show the expected information
+upon inspection, contact the group or vendor who provided it.
+
+.. _troubleshooting_keystore:
+
+Inspect JKS keystores
+"""""""""""""""""""""
+
+The Java KeyStore (JKS) system is provided by your Java installation. JKS keys
```

Comment:
Maybe this should be higher up on the page at first mention of JKS in addition or instead